shell/gpu/gpu_surface_metal.mm

```diff
@@ -80,6 +80,10 @@
 
   auto submit_callback = [this](const SurfaceFrame& surface_frame, SkCanvas* canvas) -> bool {
     TRACE_EVENT0("flutter", "GPUSurfaceMetal::Submit");
+    if (canvas == nullptr) {
+      return false;
```

Comment:
Nit: Add a DLOG like in the other case of an early return?

Reply:
added log and created an issue https://github.com/flutter/flutter/issues/56405